src/components/ListItem.jsx

```diff
@@ -44,6 +45,28 @@ export function ListItem({
 		);
 	};
 
+	useEffect(() => {
```

Comment:
I think you can write this in one conditional block with the most specific call at the top:
```
		if (dateLastPurchased !== null && numOfDaysBtwnDates(dateLastPurchased.toDate(), today) >= 60) {
			setUrgency('inactive');
		}else if (differenceOfDays <= 7) {
			setUrgency('soon');
		} else if (differenceOfDays < 30) {
			setUrgency('kind of soon');
		} else {
			setUrgency('not soon');
		}
```

Reply:
I agree. I tested this version of the code and the functionality was the same.